src/main.js

```diff
@@ -151,14 +171,106 @@ var printIssues = function(callback) {
         else return 0;
 
     });
+
+    var asyncFunctions = [];
     
     for (var k = 0; k < gCommits.length; k++) {
+        
         var commit = gCommits[k];
-        console.log("commit: " + commit.sha() + " on " + commit.moment.format("YYYY MM DD HH:mm:ss"));
+        
+        (function(commit) {
+
+            asyncFunctions.push(function(callback) {
+
+                var rootTree = commit.getTree().then(function(rootTree) {
+
+                    return rootTree.entryByPath(gTestDirectory);
+                    
+                }).then(function(entry) {
+                    
+                    if (entry.isTree()) {
+                    
+                        var repo = commit.owner();
+                        var oid = entry.oid();
+                        
+                        return repo.getTree(oid);
+                        
+                    } else {
+                        
+                        //Entry was a file, not a directory;
+                        callback(null, {
+                            moment: commit.moment,
+                            lines: 0
+                        });
+                    }
+                    
+                    return null;
+                    
+                }, function(error) {
```

Comment:
This was how I was finally able to detect if the test directory doesn't exist (ie. for early commits). For these cases, I just return 0 for lines of code.